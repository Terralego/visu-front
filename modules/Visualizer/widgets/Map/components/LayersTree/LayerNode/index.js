import LayerNode from './LayerNode';

export default LayerNode;
