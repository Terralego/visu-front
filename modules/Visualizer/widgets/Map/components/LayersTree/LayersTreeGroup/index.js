import LayersTreeGroup from './LayersTreeGroup';

export * from './LayersTreeGroup';
export default LayersTreeGroup;
