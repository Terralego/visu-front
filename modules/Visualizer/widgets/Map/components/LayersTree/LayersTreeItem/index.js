import LayersTreeItem from './LayersTreeItem';
import connectWidgetMap from '../../../connect';

export * from './LayersTreeItem';

export default connectWidgetMap(({ getLayerState, setLayerState }, { layer }) => ({
  isActive: getLayerState({ layer }).active,
  opacity: getLayerState({ layer }).opacity,
  setLayerState,
}))(LayersTreeItem);
