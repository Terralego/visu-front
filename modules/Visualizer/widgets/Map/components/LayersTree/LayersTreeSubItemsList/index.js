import LayersTreeSubItemsList from './LayersTreeSubItemsList';
import connectWidgetMap from '../../../connect';

export * from './LayersTreeSubItemsList';
export default connectWidgetMap(({ selectSublayer, getLayerState }, { layer }) => ({
  layerState: getLayerState({ layer }),
  selectSublayer,
}))(LayersTreeSubItemsList);
