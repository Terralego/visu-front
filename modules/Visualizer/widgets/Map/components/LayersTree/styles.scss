.layerNode-container {
  padding: 4px 4px 4px 8px;
  border: 1px solid #293742;
  font-size: 16px;
  font-weight: 200;
  border-radius: 0px;
}

.layerNode-header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 8px;
}

.button-more-vertical {
  transform: rotate(90deg)
}
