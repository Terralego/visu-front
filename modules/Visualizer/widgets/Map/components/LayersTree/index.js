import React from 'react';
import LayersTree from './LayersTree';
import { connectTerraFrontProvider } from '../../../../../TerraFrontProvider';

export * from './LayersTree';
export default connectTerraFrontProvider({
  renderLayersTree: 'modules.Visualizer.widgets.Map.renderLayersTree',
})(
  ({
    renderLayersTree: RenderLayersTree = LayersTree,
    ...props
  }) => <RenderLayersTree defaultRender={LayersTree} {...props} />,
);
