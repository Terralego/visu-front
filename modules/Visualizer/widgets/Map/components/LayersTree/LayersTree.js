import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Card, H4 } from '@blueprintjs/core';

import layersTreeProps from '../../LayersTreeProps';
import LayerNode from './LayerNode';
import './styles.scss';

export class LayersTree extends Component {
  static propTypes = {
    layersTree: layersTreeProps.isRequired,
    title: PropTypes.string,
  };

  static defaultProps = {
    title: 'Couches cartographiques',
  };

  onToggleChange = layer => () => {
    const { onChange } = this.props;
    const styleToApply = layer.isActive ? layer.inactive : layer.active;
    layer.isActive = !layer.isActive; // eslint-disable-line
    onChange(styleToApply);
  }

  render () {
    const { layersTree, title } = this.props;
    return (
      <Card
        className="layerstree-panel-container bp3-dark"
      >
        <H4>{title}</H4>
        {layersTree.map(layer => (
          <LayerNode
            label={layer.label}
            onToggleChange={this.onToggleChange(layer)}
            isActive={layer.isActive}
          />
        ))}
      </Card>
    );
  }
}

export default LayersTree;
