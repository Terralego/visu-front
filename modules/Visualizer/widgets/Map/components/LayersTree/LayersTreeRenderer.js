import React from 'react';
import { Card, H4 } from '@blueprintjs/core';

import LayerNode from './LayerNode';

import './styles.scss';

export const LayersTreeRenderer = ({
  layersTree,
  onToggleChange,
  isActive,
  LayersTree,
  onChange,
  getOpacity,
  onOpacityChange,
}) => (
  <Card
    className="layerstree-panel-container bp3-dark"
  >
    {layersTree.map(layer => (layer.group
      ? (
        <div key={layer.group}>
          <H4>{layer.group}</H4>
          <LayersTree layersTree={layer.layers} onChange={onChange} />
        </div>
      )
      : (
        <LayerNode
          key={layer.label}
          label={layer.label}
          onToggleChange={onToggleChange(layer)}
          isActive={isActive(layer)}
          opacity={getOpacity(layer)}
          onOpacityChange={onOpacityChange(layer)}
        />
      )))}
  </Card>
);

export default LayersTreeRenderer;
