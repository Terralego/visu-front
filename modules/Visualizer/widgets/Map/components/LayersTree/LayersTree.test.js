import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { LayersTree } from './LayersTree';


it('should render correctly', () => {
  const layersTree = [{
    label: 'foo',
    active: {
      layouts: [{
        id: 'foo',
        visibility: 'visible',
      }],
    },
    inactive: {
      layouts: [{
        id: 'foo',
        visibility: 'none',
      }],
    },
  }, {
    label: 'bar',
    active: {
      layouts: [{
        id: 'bar',
        visibility: 'visible',
      }],
    },
    inactive: {
      layouts: [{
        id: 'bar',
        visibility: 'none',
      }],
    },
  }];
  const tree = renderer.create((
    <LayersTree
      layersTree={layersTree}
    />
  )).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should change on toggle', () => {
  const onChange = jest.fn();
  const layer = {
    active: {},
    inactive: {},
  };
  const wrapper = shallow(<LayersTree onChange={onChange} layersTree={[]} />);
  const instance = wrapper.instance();
  instance.onToggleChange(layer)();
  expect(onChange).toHaveBeenCalledWith(layer.active);
  expect(instance.isActive(layer)).toBe(true);

  instance.onToggleChange(layer)();
  expect(onChange).toHaveBeenCalledWith(layer.inactive);
  expect(instance.isActive(layer)).toBe(false);
});
