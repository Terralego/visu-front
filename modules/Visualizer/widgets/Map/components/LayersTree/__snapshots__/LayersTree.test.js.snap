// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a group 1`] = `
<div
  className="bp3-card bp3-elevation-0 layerstree-panel-container bp3-dark"
>
  <div>
    <h4
      className="bp3-heading"
    >
      foo
    </h4>
    <div
      className="bp3-card bp3-elevation-0 layerstree-panel-container bp3-dark"
    >
      <div
        className="bp3-card bp3-elevation-2 layerNode-container"
        style={
          Object {
            "opacity": 0.7,
          }
        }
      >
        <label
          className="bp3-control bp3-switch"
        >
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="bp3-control-indicator"
          />
          foo
        </label>
      </div>
      <div
        className="bp3-card bp3-elevation-2 layerNode-container"
        style={
          Object {
            "opacity": 0.7,
          }
        }
      >
        <label
          className="bp3-control bp3-switch"
        >
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="bp3-control-indicator"
          />
          bar
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render correctly 1`] = `
<div
  className="bp3-card bp3-elevation-0 layerstree-panel-container bp3-dark"
>
  <div
    className="bp3-card bp3-elevation-2 layerNode-container"
    style={
      Object {
        "opacity": 0.7,
      }
    }
  >
    <div
      className="layerNode-label-container"
    >
      <label
        className="bp3-control bp3-switch"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="bp3-control-indicator"
        />
        foo
      </label>
    </div>
  </div>
  <div
    className="bp3-card bp3-elevation-2 layerNode-container"
    style={
      Object {
        "opacity": 0.7,
      }
    }
  >
    <div
      className="layerNode-label-container"
    >
      <label
        className="bp3-control bp3-switch"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="bp3-control-indicator"
        />
        bar
      </label>
    </div>
  </div>
</div>
`;
